### PR TITLE
chore(auth): legacy credential provider to use AuthOutputs instead of AmplifyConfig types

### PR DESCRIPTION
**Pull request by NikaHsn**
*Issue #, if available:*

*Description of changes:*
- legacy credential provider to use AuthOutputs instead of AmplifyConfig types
- legacy ios cognito keys to have an String field instead of CognitoUserPoolConfig
- credential store state machine to use AuthOutputs instead of AuthConfiguration

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by Equartey — APPROVED** [review]

File: packages/auth/amplify_auth_cognito/lib/src/credentials/legacy_credential_provider_ios.dart [hunk 1]
```diff
@@ -162,20 +161,21 @@ class LegacyCredentialProviderIOS implements LegacyCredentialProvider {
...
-  Future<LegacyDeviceDetails?> fetchLegacyDeviceSecrets({
-    required String username,
-    CognitoUserPoolConfig? userPoolConfig,
-  }) async {
-    if (userPoolConfig == null) return null;
+  Future<LegacyDeviceDetails?> fetchLegacyDeviceSecrets(
+    String username,
+    AuthOutputs authOutputs,
+  ) async {
+    if (authOutputs.userPoolClientId == null) return null;
```
Comment (Jordan-Nelson) [inline packages/auth/amplify_auth_cognito/lib/src/credentials/legacy_credential_provider_ios.dart:168]: *suggestion:* Remove the force casts. If you create variables for userPoolId and userPoolClientId and return early if they are null, then Dart will not require the force case.

```dart
    final userPoolId = authOutputs.userPoolId;
    final userPoolClientId = authOutputs.userPoolClientId;
    if (userPoolId == null || userPoolClientId == null) return null;
    final userPoolStorage = await _getUserPoolStorage();
    final cognitoUserKeys = LegacyCognitoUserKeys(userPoolClientId);
    final currentUserId = await userPoolStorage.read(
      key: cognitoUserKeys[LegacyCognitoKey.currentUser],
    );
    if (currentUserId == null) return null;
    final keys = LegacyDeviceSecretKeys(currentUserId, userPoolId);
```

File: packages/auth/amplify_auth_cognito/lib/src/credentials/legacy_credential_provider_ios.dart [hunk 2]
```diff
@@ -201,19 +201,20 @@ class LegacyCredentialProviderIOS implements LegacyCredentialProvider {
...
-    final cognitoUserKeys = LegacyCognitoUserKeys(userPoolConfig);
+    final cognitoUserKeys =
+        LegacyCognitoUserKeys(authOutputs.userPoolClientId!);
     final currentUserId = await userPoolStorage.read(
       key: cognitoUserKeys[LegacyCognitoKey.currentUser],
     );
-    if (currentUserId == null) return;
-    final keys = LegacyDeviceSecretKeys(currentUserId, userPoolConfig);
-    final asfKeys = LegacyAsfDeviceKeys(currentUserId, userPoolConfig);
+    if (currentUserId == null || authOutputs.userPoolId == null) return;
```
Comment (Jordan-Nelson) [inline packages/auth/amplify_auth_cognito/lib/src/credentials/legacy_credential_provider_ios.dart:215]: Same suggestion as above. Remove the force cast by creating a userPoolId var.

File: packages/auth/amplify_auth_cognito/lib/src/credentials/legacy_credential_provider_ios.dart (hunk 1 above)
Comment (Jordan-Nelson) [reply packages/auth/amplify_auth_cognito/lib/src/credentials/legacy_credential_provider_ios.dart:168]: This applies to several other locations in the PR.

File: packages/auth/amplify_auth_cognito/lib/src/credentials/legacy_credential_provider_ios.dart [hunk 3]
```diff
@@ -122,21 +122,20 @@ class LegacyCredentialProviderIOS implements LegacyCredentialProvider {
...
+      final cognitoUserKeys =
+          LegacyCognitoUserKeys(authOutputs.userPoolClientId!);
       final currentUser = await userPoolStorage.read(
         key: cognitoUserKeys[LegacyCognitoKey.currentUser],
       );
       if (currentUser != null) {
         final userPoolKeys = LegacyCognitoUserPoolKeys(
           currentUser,
-          userPoolConfig,
+          authOutputs.userPoolClientId!,
```
Comment (Jordan-Nelson) [inline packages/auth/amplify_auth_cognito/lib/src/credentials/legacy_credential_provider_ios.dart:138]: See suggestion below about remove force casts

**Review by Equartey — APPROVED** [review]

**Pull request merged**